core-bundle/src/EventListener/FilterPageTypeListener.php

```diff
@@ -34,7 +34,7 @@ public function __invoke(FilterPageTypeEvent $event): void
         }
 
         // The first level can only have root pages (see #6360)
-        if (!($currentRecord['pid'] ?? null)) {
+        if (0 === (int) $currentRecord['pid']) {
```

Comment:
Because if `$dc->activeRecord` is set, there will be a `pid` field. There was no null check before, either.

<img width="509" alt="" src="https://user-images.githubusercontent.com/1192057/176130314-a7117ac7-d6ca-421a-8cc9-4e3deb1a5c00.png">